src/algorithms/rsa.rs

```diff
@@ -194,3 +195,71 @@ fn blind<R: CryptoRngCore, K: PublicKeyParts>(
 fn unblind(key: &impl PublicKeyParts, m: &BigUint, unblinder: &BigUint) -> BigUint {
     (m * unblinder) % key.n()
 }
+
+/// The following (deterministic) algorithm also recovers the prime factors `p` and `q` of a modulus `n`, given the
+/// public exponent `e` and private exponent `d` using the method described in
+/// [NIST 800-56B Appendix C.2](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Br2.pdf).
```

Comment:
There are a set of preconditions in the "Assumptions" section that need to be checked outlined on p.113, for example that the public exponent is within 2^16..2^256

Reply:
Check public exponent before starting:
https://github.com/RustCrypto/RSA/pull/380/commits/8a259daca8c073ac7b1a5fd0cde70fbe7a371838
Verify P&Q using existing validation in the case where primes are recovered:
https://github.com/RustCrypto/RSA/pull/380/commits/de60cc32cecefa36c3b6c4409403f07738aae47c